### PR TITLE
Inlines kotlin runtime to reduce dependency complexity

### PR DESCRIPTION
**Pull request by RCHowell**
*Description of changes:*

This PR introduces a method to "include" files from the resources into the generated source tree, as well as inlines the runtime methods for the kotlin target. The kotlin target produces code which depends on a runtime. Why is that? In Smithy, you can type aliases primitives which necessarily (afaik) requires open "primitive" classes for simple OO modeling. Consider this,

```
type foo blob(10);
```

This is a simple definition which needs to create some Foo class which inherits from some Blob class, hence some builtin "runtime" types. The readers and writers also get a handful of helper methods for assertions, collection reading helpers, and IonReader/IonWriter actions like flush/finish.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: anodizer/anodizer/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+## Example
+
+```
+anodizer generate kotlin --domain example --package "org.partiql.example" ./examples/models/kitchen.ridl
```
Comment (jpschorr) [inline anodizer/anodizer/README.md:5]: I tested locally with 
```
gradle run --args="generate kotlin --domain example --package \"org.partiql.example\" ../examples/models/kitchen.ridl"
```

File: anodizer/anodizer-core/src/main/kotlin/io/github/amzn/anodizer/core/File.kt [hunk 2]
```diff
@@ -4,13 +4,20 @@ import java.io.ByteArrayInputStream
...
  *
+ * Q: More why?
+ * A: We can re-create a file hierarchy from the resources in a JAR.
+ *
+ * Q: What is wrong with this?
+ * A: For starters, there are no assertions on duplicate entities.
+ *
  * TODO tree pretty-print helper.
  */
 public class File private constructor(
```
Comment (jpschorr) [inline anodizer/anodizer-core/src/main/kotlin/io/github/amzn/anodizer/core/File.kt:23]: So. Trees of `File` are generation target for codegen. And this change adds `resource` to load and 'copy' files directly into the output.

**Review by jpschorr — COMMENTED** [review]
I notice that anodizer is hooked into the smithy build process to generate both smithy and anodizer code, and it is also vended via a basic CLI for generating only the anodizer code.

Are these both long term use-cases?

**Review by jpschorr — APPROVED** [review]

File: anodizer/anodizer-core/src/main/kotlin/io/github/amzn/anodizer/core/File.kt (hunk 2 above)
Comment (RCHowell) [reply anodizer/anodizer-core/src/main/kotlin/io/github/amzn/anodizer/core/File.kt:23]: Yes. A target produces a File (possibly a tree of dirs and files) which is a simple in-memory filesystem which a consumer can then materialize. The smithy integration walks this tree and creates the actual files through the smithy FileManifest abstraction. This ensures targets can generate arbitrarily complex output (more than just one stream of bytes) and Smithy has control of how it is actually written.

**Pull request merged**